leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -47,6 +51,7 @@
     private final Set<Endpoint> endpoints;
     private final ObservationRegistry observationRegistry;
     private final LwM2mModelProvider modelProvider;
+    private final Map<String/* registration id */, ArrayList<Request> /* pending coap Request */> pendingRequests = new HashMap<>();
```

Comment:
I'm not sure, if an additional map with key <reg id> is really required.
Ok, it "isolates the concerns" but extending the "Client" may be a alternative, which is also not too bad.